pkg/apis/serving/v1beta1/configmap.go

```diff
@@ -0,0 +1,159 @@
+	PyTorch    PredictorConfig `json:"pytorch,omitempty"`
+	ONNX       PredictorConfig `json:"onnx,omitempty"`
+}
+
+// +kubebuilder:object:generate=false
+type TransformerConfig struct {
+	// transformer docker image name
+	ContainerImage string `json:"image"`
+	// default transformer docker image version
+	DefaultImageVersion string `json:"defaultImageVersion"`
+}
+
+// +kubebuilder:object:generate=false
+type TransformersConfig struct {
+	Feast TransformerConfig `json:"feast,omitempty"`
+}
+
+// +kubebuilder:object:generate=false
+type InferenceServicesConfig struct {
+	// Transformer configurations
+	Transformers *TransformersConfig `json:"transformers"`
+	// Predictor configurations
+	Predictors *PredictorsConfig `json:"predictors"`
+	// Explainer configurations
+	Explainers *ExplainersConfig `json:"explainers"`
+}
+
+func GetInferenceServicesConfig(client client.Client) (*InferenceServicesConfig, error) {
+	configMap := &v1.ConfigMap{}
+	err := client.Get(context.TODO(), types.NamespacedName{Name: constants.InferenceServiceConfigMapName, Namespace: constants.KFServingNamespace}, configMap)
+	if err != nil {
+		return nil, err
+	}
+
+	inferenceServiceConfigMap, err := NewInferenceServicesConfig(configMap)
+	if err != nil {
+		return nil, err
+	}
+
+	return inferenceServiceConfigMap, nil
+}
+
+func NewInferenceServicesConfig(configMap *v1.ConfigMap) (*InferenceServicesConfig, error) {
+	predictorsConfig, err := getPredictorsConfigs(configMap)
+	if err != nil {
+		return nil, err
+	}
+	transformersConfig, err := getTransformersConfigs(configMap)
+	if err != nil {
+		return nil, err
+	}
+	explainersConfig, err := getExplainersConfigs(configMap)
+	if err != nil {
+		return nil, err
+	}
+	return &InferenceServicesConfig{
+		Predictors:   predictorsConfig,
+		Transformers: transformersConfig,
+		Explainers:   explainersConfig,
+	}, nil
+}
+
+func getPredictorsConfigs(configMap *v1.ConfigMap) (*PredictorsConfig, error) {
+	predictorConfig := &PredictorsConfig{}
+	if data, ok := configMap.Data[PredictorConfigKeyName]; ok {
+		err := json.Unmarshal([]byte(data), &predictorConfig)
+		if err != nil {
+			return nil, fmt.Errorf("Unable to unmarshall %v json string due to %v ", PredictorConfigKeyName, err)
+		}
+	}
+	return predictorConfig, nil
+}
+
+func getTransformersConfigs(configMap *v1.ConfigMap) (*TransformersConfig, error) {
+	transformerConfig := &TransformersConfig{}
+	if data, ok := configMap.Data[TransformerConfigKeyName]; ok {
+		err := json.Unmarshal([]byte(data), &transformerConfig)
+		if err != nil {
+			return nil, fmt.Errorf("Unable to unmarshall %v json string due to %v ", TransformerConfigKeyName, err)
+		}
+	}
+	return transformerConfig, nil
+}
+
+func getExplainersConfigs(configMap *v1.ConfigMap) (*ExplainersConfig, error) {
+	explainerConfig := &ExplainersConfig{}
+	if data, ok := configMap.Data[ExplainerConfigKeyName]; ok {
+		err := json.Unmarshal([]byte(data), &explainerConfig)
```

Comment:
yaml.DisallowUnknownFields

Reply:
is this an unmarshal option?

Reply:
Yep

Reply:
does not look like that json unmarshal has this option

Reply:
https://golang.org/pkg/encoding/json/#Decoder.DisallowUnknownFields